### ISSUE TITLE
Tanzania2011-12

### ISSUE THREAD

**Issue by robin-wang** [wengxyu1030/AIS#7]
WORKFLOW DOC

{FOR REVIEWERS}
- [x] They can run the code smoothly on the reviewer's end, where the microdata could be generated without error
- [x] Review and validate the quality checking output

{FOR CODERS}
A. General Procedural Steps

- [x] Rebase current branch based on Master, ensure your branch is the latest.
Tutorial: Rebasing your project branch onto another branch
https://docs.github.com/en/desktop/contributing-and-collaborating-using-github-desktop/syncing-your-branch#rebasing-your-project-branch-onto-another-branch

- [x] Run script without error
- [x] Pass internal quality checking (the embedded quality check result: successfully produced value for the indicators and check if there're flags that the indicator value produced is too off from the published data) 
- [x] Document as comments any notable changes/caveats (if any)
- [x] When Fully Completed Survey Edits, proceed to record hours worked: https://forms.gle/tLh7fxCMMsLobGHA6

B. Checklist and Guidance

- [x] ANC
  - Skill providers
    -  Checked in the regular express section in Programs 
    - _doctor/AMO, clinical officer, assistant clinical officer, nurse/midwife, MCH aide. [**trained traditional birth attendant/TBA** was excluded after recoding. 20210630]_ 
  - Tetanus Injection: _not available_
    -  Number of Injection Before/During Pregnancy
    -  Recall period for last injection
    -  Relevant data comes other sections
  - Blood Pressure / Blood Sample / Urine Sample: _not available_
    -  Samples (m42c/d/e) are sub-questions for "receive antenatal care" in questionnaire or not

- [x] Delivery Care
  - c_facdel /  delivery institutions  / c_hospdel / delivery institutions
    -  Consistent with the regular expression section in Programs 
    - _Formal health facilities:  dispensary, district hospital, health centre, private dispensary, private health centre, private specialised hospital, referral/spec. hospital, regional hospital, religious dispensary, religious district hospital, religious referral/spec. hospital, village health post. [**delivered in the way to the hospital** was excluded after recoding 20210630]_
  - c_skin2skin: _not available_
    -  Check if available; if, then note which variable
  - m15 skip condition: _Checked survey report AIS11, p.268, m15 does not contain skip conditions._
    -  Usually delivery location is a filter question for c_caesarean(m17) / stay(m61) / mother or child PNC; is it m15, if not which one, and list the conditions used for m15  if pattern is peculiar (normally (m15 = 11, 12 etc))

- [x] PNC: _not available_
  - child pnc
    -  See if child pnc is only avaliable when (m15=11,12,96) or (m15!=11,12,96)
    -  Check if Child PNC is designed to be from full sample, i.e not a subquestion within fertility section. If a subquestion, don't generate c_pnc_any and c_pnc_eff
   
- [x] Child Vaccination (search receive/month in birth.dta. if unknown vaccine pops up, search definition in DHS report) : _not available_ 
  - BCG
  - DTP/DPT, DTaP, DTwP or pentavalent 1-3 against diphtheria, tetanus and pertussis
  - Polio 1-3 (IPV, OPV)
  - Measles/MMR
  - Other vaccines, if any,

- [x] Child Illness
  - 20210510 For Providers, focus on the skilled providers rather than all
  - Diarrhea treatment provider: _not available_
          - Check provider list being consistent with existing list (if not make changes in program)
  - Fever/Ari1/Ari2 treatment provider
          - Provider list being consistent with existing list
          - _public referral/spec. hospital, public regional hospital, public district hospital, public health centre, public dispensary, public village health post, private specialised hospital, private health centre, private dispensary, religious referral/spec. hospital, religious district hospital, religious health centre, religious dispensary, NGO_
  - Fever/Fever Treatment/Cough/Cough Treatment from Full Sample: _yes_
  - c_diarrhea_med, medicine and formal meds to treat are survey-specific. 20210510 focus on special medicines: _not available_
  - c_diarrhea_medfor, medicine and formal meds to treat are survey-specific. 20210510 focus on special medicines: _not available_
  - Validate c_ari can be generated or not: _not available_

- [x] Child Mortality
  - General Observations

- [x] Sleeping Under Bednet
  - Adhere to program's logic, but please list any specific bednet related observations (tricky cases for presumed treated or the like): _ml0 1= only treated bednets_

- [x] Child Anthropometrics
  - General Observations (No need to generate "Wasting")

- [x] Reproductive Health
  - Contraceptive, check modern methods: _not available_
  - In Need - most updated version of logics in program (temporarily no actions needed)

- [x] Female Anthropometrics and Cancer Screening: _not available_	
  - Mammogram (X-ray, mamm...)
  - Pap Smear (Cervical vaginal cytology)
  - Note recall indicators for Mamm or Pap Smear (for example, s1012n used as filter for s1012m)

- [x] Hospitalisation, NCD & HIV
  - Hospitalisation, general observations: _not available_			
  - Hypertension treatment , blood pressure measure in the past or during the survey, for example check hm.dta/ind.dta/men.dta (generate a_bp_treat a_bp_sys a_bp_dial a_bp_meas a_hi_bp140 a_hi_bp140_or_on_med from ind.dta & men.dta by combining them): _not available_

    - Note from WB team for reference: "don't worry if you cannot find men.dta, not all surveys have it. if it is collected from ind.dta and men.dta seperately (some survey have female & male questionnarie, collecting these data seperately), you will need to combine these 2 dataset, generating these indicators and merge them back to hm.dta. If only from ind.dta or men.dta, i.e. only for female or male, don't generate these variables. If a new/survey-specific datafile occurs, it is highly likely that it contains info. for 13.do.  "
    
  - Diabetes treatment/measure in the past (for example, a_diab_treat): _not available_
  - HIV, general observations: _hiv03 blood test result: hiv negative, hiv positive_
    - HIV questionnaire might exsit in Section 4 of "HEALTH ISSUES QUESTIONNAIRE". If they don't ask hiv test results, they are not helpful. but please document these observations.

- [x] Cross-validate statistics in survey report, if applicable.
     - _validated with the survey report SR196, the report indicates the mean HIV-positive in 2011-12 should be 5.1 in total, while the dataset suggests 0.457._
     - _AIS11, p.149 "percentage for whom advice or treatment was sought from a health facility, provider, or pharmacy among children under age 5 with fever" = 0.773, the dataset suggests 0.551._
     - _other indicators are not discussed in the survey report._

- [x] Reference the NAME-SURVEY branch when your edits are finalised.
   - Refer to the github working branch created for [Name-SurveyName]

- [x] Update micro data in Onedrive

- [x] Examine Quality Checking Output Files
  - [x] Successfully produced the indicators
  - [x] Check if there's a flag that the indicator value produced is too off from the published data
        - _no more flags after recoding._

**Comment by yining-sun** [comment]
@robin-wang @wengxyu1030 

Dear both, 

I have initiated the pull request for #21 and uploaded the micro-data. However, when I checked the formal treatment provider, I found several items kept under the current codes should not be considered formal. Please see the bolded items above. Let me know if you would like to leave it like that or make changes. 

Best, 
Yining

**Comment by wengxyu1030** [comment]
Dear Yining, 

Thanks for the update, overall it looks great! 

I agree that the providers you listed in bold are controversial to be considered formal providers. I would suggest in cases like this, you could take actions on:

1. Check the country report (https://www.dhsprogram.com/methodology/survey/survey-display-393.cfm) and see if there's a definition of skilled providers. 
2. If in 1 you can not see the definition. Try to tab the providers by type, and guestimate the cause of difference to DHS (as Xian suggested in the conversation: https://github.com/wengxyu1030/MIS/pull/9#issuecomment-864194441)

If none of this works, let's document the efforts made in the issue and park this conversation for future references. 

Regards,
Aline

**Comment by yining-sun** [comment]
@wengxyu1030 @robin-wang 

Dear Aline, 

Thanks so much for the comments. They are very helpful. 

1. I have excluded the "Delivery in the way to the hospital" from formal delivery list. 

2. Regarding the skilled provider list: 
- Quality control data: if keep the bolded item "trained traditional birth attendant". 
<img width="748" alt="Screen Shot 2021-06-30 at 10 43 48 AM" src="https://user-images.githubusercontent.com/60246203/124007348-25f0fb00-d990-11eb-98a8-60f5ecf64cb3.png">

- Quality control data: if excluded the bolded item "trained traditional birth attendant", there is no more flags of skilled provider. 
<img width="756" alt="Screen Shot 2021-06-30 at 10 58 44 AM" src="https://user-images.githubusercontent.com/60246203/124009234-36a27080-d992-11eb-82f2-794fea30544e.png">

Therefore, I initiated another pull request  #24  and updated the microdata in the onedrive folder. Let me know if you have any questions or comments.

Many thanks, 
Yining
